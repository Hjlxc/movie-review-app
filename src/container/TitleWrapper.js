import React from 'react';

import {
  HorizontalFlexWrapper,
  VerticalFlexWrapper,
  BoldWrapper,
  Gap,
} from './styledComponent';

const verticalAlign = 'vertical';

export default function({ title, children, align = verticalAlign }) {
  const Wrapper =
    align === verticalAlign ? VerticalFlexWrapper : HorizontalFlexWrapper;
  return (
    <Wrapper>
      {title && <BoldWrapper>{title}</BoldWrapper>}
      {title && align !== verticalAlign && <Gap />}
      {children}
    </Wrapper>
  );
}
