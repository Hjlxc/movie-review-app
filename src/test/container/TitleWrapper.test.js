import React from 'react';
import renderer from 'react-test-renderer';
import { render, cleanup } from '@testing-library/react';

import TitleWrapper from '../../container/TitleWrapper';

const childComponent = <div>Wrapper Child</div>;

test('matches snapshot vertical', () => {
  const testRenderer = renderer.create(
    <TitleWrapper>{childComponent}</TitleWrapper>
  );
  const tree = testRenderer.toJSON();
  expect(tree).toMatchSnapshot();
});

test('matches snapshot horizontal', () => {
  const testRenderer = renderer.create(
    <TitleWrapper align="horizontal">{childComponent}</TitleWrapper>
  );
  const tree = testRenderer.toJSON();
  expect(tree).toMatchSnapshot();
});

test('matches snapshot vertical with title', () => {
  const testRenderer = renderer.create(
    <TitleWrapper title="Wrapper Title">{childComponent}</TitleWrapper>
  );
  const tree = testRenderer.toJSON();
  expect(tree).toMatchSnapshot();
});

test('matches snapshot horizontal with title', () => {
  const testRenderer = renderer.create(
    <TitleWrapper title="Wrapper Title" align="horizontal">
      {childComponent}
    </TitleWrapper>
  );
  const tree = testRenderer.toJSON();
  expect(tree).toMatchSnapshot();
});

describe('test correctly render component', () => {
  beforeEach(cleanup);
  it('Should render children', () => {
    const { getByText } = render(<TitleWrapper>{childComponent}</TitleWrapper>);
    expect(getByText('Wrapper Child')).not.toBeNull();
  });
  it('Should render title', () => {
    const { getByText } = render(
      <TitleWrapper title="Wrapper Title">{childComponent}</TitleWrapper>
    );
    expect(getByText('Wrapper Title')).not.toBeNull();
    expect(getByText('Wrapper Child')).not.toBeNull();
  });
});
