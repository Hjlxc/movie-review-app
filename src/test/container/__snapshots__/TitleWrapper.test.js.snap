// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot horizontal 1`] = `
<div
  className="sc-AxirZ hlnYRa"
>
  <div>
    Wrapper Child
  </div>
</div>
`;

exports[`matches snapshot horizontal with title 1`] = `
<div
  className="sc-AxirZ hlnYRa"
>
  <div
    className="sc-AxhUy sc-AxheI bvzwzp"
  >
    Wrapper Title
  </div>
  <div
    className="sc-AxgMl cEDzwU"
  />
  <div>
    Wrapper Child
  </div>
</div>
`;

exports[`matches snapshot vertical 1`] = `
<div
  className="sc-AxiKw jWoTjM"
>
  <div>
    Wrapper Child
  </div>
</div>
`;

exports[`matches snapshot vertical with title 1`] = `
<div
  className="sc-AxiKw jWoTjM"
>
  <div
    className="sc-AxhUy sc-AxheI bvzwzp"
  >
    Wrapper Title
  </div>
  <div>
    Wrapper Child
  </div>
</div>
`;
